ripngd/ripng_nb_config.c

```diff
@@ -129,9 +129,13 @@ int ripngd_instance_allow_ecmp_modify(struct nb_cb_modify_args *args)
 		return NB_OK;
 
 	ripng = nb_running_get_entry(args->dnode, NULL, true);
-	ripng->ecmp = yang_dnode_get_bool(args->dnode, NULL);
-	if (!ripng->ecmp)
+	ripng->ecmp = yang_dnode_get_uint8(args->dnode, NULL);
```

Comment:
Do you mean to set `ripng->ecmp` to `MIN(yang_dnode_get_uint8(args->dnode, NULL), ripng->ecmp)`? But don't understand why or I'm missing something :)

Reply:
yes.  here is why, suppose FRR is compiled with 256 way ecmp, zebra is started with 4 way ecmp and rip has a config file change that says use 8 way ecmp.  ripngd should use 4 way ecmp since it will have received the 4 way ecmp as part of the capabilities.  Zebra might have already been connected to by this point in time and there is no point in overriding what zebra is telling us and giving it data it cannot use

Reply:
But this way we always drop the higher value. If we have `allow-ecmp 2`, but trying to increase it to `allow-ecmp 4`, this will never happen and 2 will be returned.

Reply:
We have this validation here https://github.com/FRRouting/frr/pull/13535/files#diff-32583cf3497b45df469a41c69215161d666923cfacc89218fa0ff29bd0f8c24bR233

Reply:
then we should save the value received from zebra and the value received from the cli and then make the right decision

Reply:
Should be fixed now.